source/core.py

```diff
@@ -419,6 +419,20 @@ def _handleNVDAModuleCleanupBeforeGUIExit():
 	brailleViewer.destroyBrailleViewer()
 
 
+def _initializeObjectCaches():
+	import api
+	import NVDAObjects
+	import winUser
+	foregroundObject = NVDAObjects.window.Window(windowHandle=winUser.getForegroundWindow())

```

Comment:
Although I agree with this approach in principle, consider the note in the [GetForegroundWindow](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getforegroundwindow):
"The foreground window can be NULL in certain circumstances, such as when a window is losing activation."
So, it is possible that we won't be able to actually create the foreground object. It is quite rare, but I've definitely seen it.

Reply:
GetForegroundWindow will always become valid eventually, but you may need to poll it for a second or two while it is NULL.

Reply:
https://github.com/nvaccess/nvda/pull/14301#pullrequestreview-1155899222

>GetForegroundWindow will have to continuously polled to ensure it is not NULL. This polling should be allowed to take as long as it needs... E.g. 10, 20 seconds... the setting of these objects can not fail.
Though this situation is extremely rare, and most likely will correct itself within a second or two.

Should we set a timeout and fail the start up after X seconds?

Reply:
You could probably fail after say 20 seconds or so I guess. And log a critical message. Otherwise those debugWarnings may go unnoticed with users just complaining of a freeze on startup.